app/controllers/puppetclass_lookup_keys_controller.rb

```diff
@@ -4,8 +4,9 @@ class PuppetclassLookupKeysController < LookupKeysController
   before_action :setup_search_options, :only => :index
 
   def index
-    @lookup_keys = resource_base_search_and_page(:param_classes).smart_class_parameters
-    @puppetclass_authorizer = Authorizer.new(User.current, :collection => @lookup_keys.map {|key| key.param_class.try(:id)}.compact.uniq)
+    @lookup_keys = resource_base_search_and_page.distinct.smart_class_parameters.preload(:lookup_values)
+    environment_classes = EnvironmentClass.where(puppetclass_lookup_key_id: @lookup_keys.map(&:id)).select(:puppetclass_id, :puppetclass_lookup_key_id).distinct.preload(:puppetclass)
+    @lookup_keys_to_class = Hash[environment_classes.map { |environment_class| [environment_class.puppetclass_lookup_key_id, environment_class.puppetclass]}]
```

Comment:
`.smart_class_parameters` already joins environment_class, so I don't think we need to toad the environment classes again - we should already have access to `environment_class.puppetclass_id` in `@lookupkeys`

Reply:
`.smart_class_parameters` supplies the `.joins` statement, but it does not add the environment_classes relation to the results. (We can't add it to the result, since it will mess up the pagination).
The new code preloads lookup_values with lookup keys and preloads puppetclasses with stripped down environment classes. I need the environment_class object for the lookup key connection.